scripts/debug_tools/reduce_ctu_crash.py

```diff
@@ -0,0 +1,625 @@
+                arg_vect[option_index + separate + 1:]
+        else:
+            break
+
+    return arg_vect
+
+
+def __remove_dependency_generation(command):
+    """
+    Remove dependency generating flags from the compilation command.
+    Expects command as a list and returns a modified list.
+    """
+    command = __eliminate_argument(command, '-MM')
+    command = __eliminate_argument(command, '-MF', True)
+    command = __eliminate_argument(command, '-MP')
+    command = __eliminate_argument(command, '-MT', True)
+    command = __eliminate_argument(command, '-MQ', True)
+    command = __eliminate_argument(command, '-MD')
+    command = __eliminate_argument(command, '-MMD')
+    command = __eliminate_argument(command, '-MJ', True)
+    command = __eliminate_argument(command, '-MV')
+    return command
+
+
+def get_preprocess_cmd(comp_cmd, repro_dir, filename, clang_path):
+    """
+    From a gcc compile command in string format, create a
+    clang preprocessing command in list format.
+    """
+    preproc_cmd = __remove_dependency_generation(comp_cmd.split(' '))
+    preproc_cmd[0] = clang_path
+    preproc_cmd.insert(1, '-E')
+
+    out_ind = preproc_cmd.index('-o')
+    preproc_cmd[out_ind + 1] = os.path.join(repro_dir, filename)
+
+    # FIXME: CodeChecker API that converts a gcc cmd to a clang cmd?
+    if '-Wno-maybe-uninitialized' in preproc_cmd:
+        preproc_cmd.remove('-Wno-maybe-uninitialized')
+
+    return preproc_cmd
+
+
+def preprocess(cmd, output_dir, clang):
+    """
+    Assemble and run the preprocessing command for a given compilation command.
+    """
+    preproc_cmd = get_preprocess_cmd(cmd['command'],
+                                     output_dir,
+                                     os.path.basename(cmd['file']),
+                                     clang)
+    prepare_cmds.execute(preproc_cmd, cwd=cmd['directory'])
+
+
+def save_new_analyzer_cmd(output_dir, args):
+    analyze_sh = os.path.join(output_dir, 'analyze.sh')
+    with open(args.analyzer_command, 'r') as old:
+        with open(analyze_sh, 'w') as new:
+            triple_arch = prepare_cmds.get_triple_arch(old.name)
+            new_cmd = get_new_analyzer_cmd(args.clang, old.read(),
+                                            output_dir, triple_arch)
+            analyzed_file = new_cmd.split(" ")[-1]
+            analyze_std_flag = get_std_flag(new_cmd)
+            new.write(new_cmd)
+
+    st = os.stat(analyze_sh)
+    os.chmod(analyze_sh, st.st_mode | stat.S_IEXEC)
+    return analyze_sh, analyzed_file, analyze_std_flag
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(
+        description='Reduces reproduction files for CTU assertions '
+        'based on the repro zip file generated by CodeChecker. '
+        'This script requires CodeChecker to be added to PATH '
+        'and its environment sourced. You can specify which Clang to use '
+        '(it is taken from PATH by default).')
+
+    # Required arguments.
```

Comment:
you can set it explicitly that this argument is [required](https://docs.python.org/3/library/argparse.html#required)